### PR TITLE
Modified _calculateAllocation to avoid funds reduction due to rounding errors.

### PR DESCRIPTION
**Pull request by takadr**
### What I did
- Revised the logic of _calculateAllocation to address the issues that led to loss of funds when allocatedAmount > totalRaised
- Implemented max restraint for allocatedAmount to prevent potential overflow within _calculateAllocation
- Implemented max restraint for minRaisedAmount to prevent the situation where minRaisedAmount > max of totalRaised

### CODE REVIEW

File: contracts/FactoryV1.sol [hunk 1]
```diff
@@ -66,6 +66,16 @@ contract FactoryV1 is ReentrancyGuard, Ownable {
             "Having an event without tokens are not permitted."
         );
 
+        require(
+            allocatedAmount < 1e50,
```
Comment (0xHaku) [inline contracts/FactoryV1.sol:70]: 定数化してください

File: contracts/SaleTemplateV1.sol [hunk 2]
```diff
@@ -135,27 +142,17 @@ contract SaleTemplateV1 is ISaleTemplateV1, ReentrancyGuard {
...
-            tda/tp can be zero
-        */
-
-        /* 
-            For a sale such that accumulates many ETH, and selling token is a few (e.g., Art NFTs),
-            if the claimer depoited only a few ETH, then allocation is 0 and will be refunded.
-            That would be acceptable behavior.
+            us<tr is always true and so us/tr is always zero
+            aa can be 1 to 10^50-1
+            (us x aa) can overflow
```
Comment (0xHaku) [inline contracts/SaleTemplateV1.sol:151]: uint256のmaxは1e77より大きいのでallocatedAmountとminRaisedAmountはless than equalにしても大丈夫です

File: contracts/FactoryV1.sol (hunk 1 above)
Comment (takadr) [reply contracts/FactoryV1.sol]: TOKEN_UPPER_BOUND, ETH_UPPER_BOUNDを追加しました。

File: contracts/SaleTemplateV1.sol (hunk 2 above)
Comment (takadr) [reply contracts/SaleTemplateV1.sol]: Less than or equal toに制約を変更しました。

**Pull request merged**